{
  try {
    return run(query);
  }
 catch (  final Exception ex) {
    ex.printStackTrace();
    final AssertionError err=new AssertionError("Query failed:\n" + query);
    err.initCause(ex);
    throw err;
  }
}
